### PR TITLE
Disable OCS api when in maintenance mode

### LINKED ISSUE #15456: Provisioning API: still working with maintenance mode on
### Steps to reproduce

Turn on the maintenance mode from the command-line
Add an user using the provisioning API: `curl -u user:password -X POST http://server:port/ocs/v1.php/cloud/users -d userid=foo -d password=11111`

### Expected behaviour

An error is thrown due to the maintenance mode
### Actual behaviour

The user is added
### Server configuration

Operating system: ubuntu 14.04

Web server: apache 2.4.7

Database: mysql

PHP version: 5.5.4

ownCloud version: 8.0.3RC1 EE

Updated from an older ownCloud or fresh install: fresh install

### PR DESCRIPTION
**Pull request by nickvergessen**
Fix #15456, #15436

@MorrisJobke @PVince81 @jvillafanez 

@DeepDiver1975 should backport all the way down to 7 I think?

### CODE REVIEW

File: ocs/v1.php [hunk 1]
```diff
@@ -27,7 +27,7 @@
 
 require_once '../lib/base.php';
 
-if (\OCP\Util::needUpgrade()) {
+if (\OCP\Util::needUpgrade() || \OC::$server->getSystemConfig()->getValue('maintenance', false)) {
```
Comment (nickvergessen) [inline ocs/v1.php]: Must be changed on backport because `getSystemConfig()` is 8+ only

**Comment by PVince81** [comment]
I just tried with the following call:
`curl -D - http://root:admin@localhost/owncloud/ocs/v1.php/apps/files_sharing/api/v1/shares`

On master: it returns 200 with empty content. (without maintenance mode it returns XML)
On this branch: it returns 503 status code and also proper XML with the status code

**Comment by PVince81** [comment]
Tested, code looks good :+1:

**Comment by DeepDiver1975** [comment]
:+1:

**Comment by PVince81** [comment]
@karlitschek please confirm backport

**Pull request merged**